lib/generator/template_values.rb

```diff
@@ -31,8 +31,40 @@ def cases_require_name
       Files::GeneratorCases.filename(exercise_name)
     end
 
+    def test_cases
+      if test_cases_proc?
+        test_cases_proc.call(canonical_data.to_s)
+      else
+        test_cases_with_index
+      end
```

Comment:
There's another object hiding in here.
Replace all this with:
```
test_cases_foobar.call(canonical_data.to_s)
```
Then work out what test_cases_foobar is.